### ISSUE TITLE
Sonic CD (200940)

### ISSUE THREAD

**Issue by lineber** [ValveSoftware/Proton#455]
OS: Fedora release 28 (Twenty Eight) x86_64
Kernel: 4.17.14-202.fc28.x86_64 
Resolution: 1920x1080, 1280x1024 
DE: GNOME 3.28.3 
WM: GNOME Shell 
WM Theme: Adwaita 
Theme: Adwaita [GTK2/3] 
CPU: Intel Pentium G4620 (4) @ 3.700GHz 
GPU: NVIDIA GeForce GTX 1060 3GB 
Nivida Driver Version: 396.45
Memory: 4540MiB / 15492MiB 

Flashes black, then crashes. I tried changing the layer to win98 still no luck.

**Comment by Zero86Sk** [comment]
Same here i tried the launch options:
PROTON_USE_WINED3D11=1 %command%
PROTON_NO_D3D11=1 %command%

(not sure if im doing it right)
but nothing works
i dont know if you can combine the commands

my log
[steam-200940.log](https://github.com/ValveSoftware/Proton/files/2320128/steam-200940.log)

my PC:
https://gist.github.com/Zero86Sk/d629f657c1ae861c36657cdf238a7765

**Comment by aqxa1** [comment]
This can be worked around by installing the native version of d3dcompiler_43:

WINEPREFIX=(steam library location)/steamapps/compatdata/200940/pfx winetricks d3dcompiler_43

**Comment by lineber** [comment]
+thirdeyefunction That is correct. It works. Thanks.

**Comment by The-King-of-Toasters** [comment]
Also confirming that @thirdeyefunction's trick works.

**Comment by Houtworm** [comment]
Game does not run, Crashes immediately in Steam Runtime
Arch Linux
GNOME 3.28.2
Nvidia 396.54

I tried installing winetricks, ran winetricks d3dcompuler_43 and set the launch command but that did not work for me, I guess i am doing something wrong

I tried 
WINEPREFIX=/home/houtworm/.local/share/Steam/steamapps/compatdata/200940/pfx winetricks d3dcompiler_43
WINEPREFIX=/home/houtworm/.local/share/Steam/steamapps/compatdata/200940/pfx/drive_c/windows/system32/d3dcompiler_43.dll
I also tried to access it in the .wine folder with the launch command
I also tried placing the DLL file in the pfx folder and accessing that with the launch command,

None seem to work, I am new to wine, So i probably did something wrong.

Added my terminal output in a gist 
https://gist.github.com/Houtworm/03d2b649e26659167813367083f0b7bd

**Comment by kode54** [comment]
I installed the game today, and tried to run it with Proton 3.7 beta 4, and it just sits in memory and does nothing. I have to pgrep -i soniccd and kill it. Installing d3dcompiler_43 in the prefix has no effect.

System stats:

https://gist.github.com/kode54/3ff952f39b407d6f806a6779838f5335

**Comment by bruno-fs** [comment]
I made it work after installing the DXSETUP.EXE included within the game directory on the game wineprefix.

[System stats](https://gist.github.com/varnion/73786ff3a9b41ca2c524bdfcf61cffb7)

I played an entire bad ending run without issues.

Not a fan of this approach though cuz it require running the installation wizard... an scriptable aproach would be preferable, but I'm new to wine... 

This is the path for the file:
 `C:\Program Files\Steam\steamapps\common\sonic cd\directx\DXSETUP.exe`
(is this exe somehow related to d3dcompiler_43? I got the game running before bumping on the solution posted here)

**Comment by Merkosas** [comment]
Yo instale Directx9 por wine y luego copie las carpetas **SYSTEM32 Y SYSWOW64** que estan en 
**home/.wine/drive_c/windows** y las puse en **home/.steam/steam/steamapps/compatdata/200940/pfx/drive_c/windows** 

Para ver las carpetas en **home** hay que hacer Ctrl+H y Listo

**Comment by Merkosas** [comment]
Google translate:
I installed Directx9 by wine and then I copied the **SYSTEM32 and SYSWOW64** folders that are in
**home/.wine/drive_c/windows** and put them in **home/.steam/steam/steamapps/compatdata/200940/pfx/drive_c/windows**

To see the folders in home you have to do Ctrl + H and Done

**Comment by jameme0** [comment]
OS: Arch Linux x64
CPU: Ryzen 5 1600
GPU: RX 480
Kernel: 4.18.12-arch1-1-ARCH 
DE: XFCE 4.12
Mesa: 18.2.2-1
Proton: Default as of 13/10/18

I ran the Direct X redist installer in the wineprefix and the game started. I haven't tested the video resolution config tool. My DualShock 4 appears to work as well.

**Comment by bruno-fs** [comment]
Had to reinstall on my PC and I avoided all the manual instalation running this
```
export STEAMLIBRARY="${HOME}/.steam/steam"
export PROTONPATH="${STEAMLIBRARY}/steamapps/common/Proton 3.7"
export WINEPATH="${PROTONPATH}/dist/bin/wine"
export WINEPREFIX="${STEAMLIBRARY}/steamapps/compatdata/200940/pfx"
winetricks d3dcompiler_43
```

**Comment by TheSandBoxMKG** [comment]
I tried both methods, installing the game's `DXSETUP.EXE` and using winetricks's `d3dcompiler_43
` but neither works, game still goes black and exits.

OS: Debian 9 Stretch
CPU: 4 Intel(R) Core(TM) i5-7200U CPU @ 2.50GHz
GPU: Intel Corporation HD Graphics 620 
Kernel: Linux version 4.9.0-8-amd64 (debian-kernel@lists.debian.org) (gcc version 6.3.0 20170516 (Debian 6.3.0-18+deb9u1) ) #1 SMP Debian 4.9.130-2 (2018-10-27)
Desktop Environment: Xfce 4.12
Proton: 3.7 (The default one)

**Comment by TheSandBoxMKG** [comment]
System Information: [Gist](https://gist.github.com/TheSandBoxMKG/35259fffe98886e013022a6ab76981c8)
Proton version: ![imagen](https://user-images.githubusercontent.com/27746283/49050077-7db79c00-f1c0-11e8-88ed-457c29f2538f.png)


[steam-200940.log](https://github.com/ValveSoftware/Proton/files/2617697/steam-200940.log)

Can say that even with [this](https://www.reddit.com/r/wine_gaming/comments/8r6low/guide_how_to_install_net_45_on_64bit_prefixes/) guide, I can't initialize Video Setup as it fails to launch .NET framework 4.7.2
![imagen](https://user-images.githubusercontent.com/27746283/49048900-65914e00-f1bb-11e8-9a66-fa3f6ccb4553.png)

Though the normal game runs fine aside from the graphical glitches and the stuttering audio.

**Comment by Thesaurus-Kit** [comment]
I installed  winetricks d3dcompiler_43 and the game didn't launch
Than I installed directx which is in game files and the Game launched
But there is another item called video setup which does not work (crashes on startup) and I have no idea what this app does.
![screenshot from 2019-01-01 01-03-04](https://user-images.githubusercontent.com/9348998/50568339-5759c300-0d61-11e9-9164-6f619676ab4c.png)

I think bundled directx should be installed automatically like it does in most of windows games.

**Comment by aqxa1** [comment]
As an update, it needs both d3dx9_43 and d3dcompiler_43 (but only the latter in earlier versions of Proton for whatever reason).

So:

`WINEPREFIX=(steam library location)/steamapps/compatdata/200940/pfx winetricks d3dx9_43 d3dcompiler_43`

**Comment by khanson679** [comment]
Installing DirectX installer included in the game did the trick for me as well, running `WINEPREFIX=.../steamapps/compatdata/200940/pfx wine DXSETUP.exe` from the "directx" folder. (I realize now that my system Wine was used in this step, but it doesn't seem to have mattered.) Winetricks does not seem to be necessary. I'm using Proton 4.2.7.

**Comment by AniLeo** [comment]
I'm subscribing here since I just got this issue

Can confirm this is still an issue as of Proton 5.13-1

- Play Sonic CD
`Assertion failed: compare_hlsl_types(arg1->data_type, arg2->data_type), file ../../../../proton/wine/dlls/d3dcompiler_47/../d3dcompiler_43/hlsl.y, line 627`
- Video Setup
`System.TypeLoadException: Could not load type of field 'soniccd_setup.Form1:availableModes' (3) due to: Could not load file or assembly 'Microsoft.DirectX.Direct3D, Version=1.0.2902.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' or one of its dependencies.`

**Comment by CarbonNotes** [comment]
With Proton 7.0-2, the game runs but only seems to display a black screen.

**Comment by Blisto91** [comment]
@CarbonNotes Can you post a proton log? :)

**Comment by CarbonNotes** [comment]
Yes I can. This is starting the game, letting it sit until I believe the Main Menu music started playing, then hitting ALT+F4:

https://gist.github.com/CarbonNotes/e724c0e2af583ef6fe7668f3d27a06bb

**Comment by Blisto91** [comment]
Hmm yes i see. Both with WINED3D and dxvk.
It works if i launch with proton GE and still works when switching back to regular proton after having run it through GE once.

**Comment by CarbonNotes** [comment]
I can confirm that after launching the game once with Proton-GE 7.16, the game now seems to work in Proton 7.0-2.
